### DISCUSSION TITLE
pod ImagePullBackOff error

### DISCUSSION THREAD

**Discussion by minorabanggu** [cloudforet-io/community#129] [General]
Hello. There is an error while installing cloudforet.

An error occurs when a certain number of pods are installed. (ImagePullBackOff)

As for the error, it seems that docker hub does not receive the corresponding version.

======================================================================

![스크린샷 2023-07-18 오후 3 29 52](https://github.com/cloudforet-io/cloudforet/assets/139847368/182b976f-5c57-4cff-b9fa-e608b7473de3)
- pod status

Name:             cost-analysis-579bd79878-8s4bs
Namespace:        spaceone
Priority:         0
Service Account:  default
Node:             minikube/192.168.49.2
Start Time:       Tue, 18 Jul 2023 05:53:11 +0000
Labels:           app.kubernetes.io/instance=cloudforet
                  app.kubernetes.io/managed-by=Helm
                  app.kubernetes.io/name=cost-analysis
                  app.kubernetes.io/version=1.x.y
                  helm.sh/chart=cost-analysis-1.3.14
                  pod-template-hash=579bd79878
                  spaceone.service=cost-analysis
Annotations:      spaceone.application-flag: db7908b4d03cdddc55d9ed33d6813c340a9f51727b8ccb61ff1f1ea5be3a3d85
                  spaceone.database-flag: 52133ddfcd583280388a8bbfb67a417b6b660f75bc63ec1b01beec811d4498ef
                  spaceone.deployment.tool: helm
                  spaceone.shared-flag: 52133ddfcd583280388a8bbfb67a417b6b660f75bc63ec1b01beec811d4498ef
Status:           Pending
IP:               10.244.0.92
IPs:
  IP:           10.244.0.92
Controlled By:  ReplicaSet/cost-analysis-579bd79878
Containers:
  cost-analysis:
    Container ID:   
    Image:          spaceone/cost-analysis:1.12.dev8
    Image ID:       
    Port:           50051/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       ImagePullBackOff
    Ready:          False
    Restart Count:  0
    Environment:
      SPACEONE_CONFIG_FILE:  /opt/spaceone/cost-analysis/config/config.yaml
    Mounts:
      /etc/localtime from timezone (rw)
      /opt/spaceone/cost-analysis/config/application.yaml from application-conf (ro,path="application.yaml")
      /opt/spaceone/cost-analysis/config/config.yaml from default-conf (ro,path="config.yaml")
      /opt/spaceone/cost-analysis/config/database.yaml from database-conf (ro,path="database.yaml")
      /opt/spaceone/cost-analysis/config/shared.yaml from shared-conf (ro,path="shared.yaml")
      /var/log/spaceone from log-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-btsp9 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  default-conf:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cost-analysis-conf
    Optional:  false
  database-conf:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cost-analysis-database-conf
    Optional:  false
  shared-conf:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      shared-conf
    Optional:  false
  application-conf:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      cost-analysis-application-grpc-conf
    Optional:  false
  timezone:
    Type:          HostPath (bare host directory volume)
    Path:          /usr/share/zoneinfo/Asia/Seoul
    HostPathType:  
  log-volume:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  kube-api-access-btsp9:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason       Age                  From               Message
  ----     ------       ----                 ----               -------
  Normal   Scheduled    35m                  default-scheduler  Successfully assigned spaceone/cost-analysis-579bd79878-8s4bs to minikube
  Warning  FailedMount  35m (x2 over 35m)    kubelet            MountVolume.SetUp failed for volume "default-conf" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  35m (x2 over 35m)    kubelet            MountVolume.SetUp failed for volume "database-conf" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  35m (x3 over 35m)    kubelet            MountVolume.SetUp failed for volume "application-conf" : failed to sync configmap cache: timed out waiting for the condition
  Warning  FailedMount  35m (x4 over 35m)    kubelet            MountVolume.SetUp failed for volume "shared-conf" : failed to sync configmap cache: timed out waiting for the condition
  Warning  Failed       34m (x2 over 35m)    kubelet            Error: ImagePullBackOff
  Normal   Pulling      34m (x3 over 35m)    kubelet            Pulling image "spaceone/cost-analysis:1.12.dev8"
  Warning  Failed       34m (x3 over 35m)    kubelet            Failed to pull image "spaceone/cost-analysis:1.12.dev8": rpc error: code = Unknown desc = Error response from daemon: manifest for spaceone/cost-analysis:1.12.dev8 not found: manifest unknown: manifest unknown
  Warning  Failed       34m (x3 over 35m)    kubelet            Error: ErrImagePull
  Normal   BackOff      51s (x146 over 35m)  kubelet            Back-off pulling image "spaceone/cost-analysis:1.12.dev8"

**Comment by GeunjeLEE** [comment]
Hi @minorabanggu.
Thank you for your reporting.

As we said in the email,
We have released a new version of the Cloudforet helm chart.

Please reinstall with the new version.

---

### uninstall old resources
```
$ helm uninstall cloudforet -n spaceone
```

### upgrade helm chart
```
$ helm repo update
```

### Ensure that the chart version is updated(1.11.11)
```
$ helm search repo cloudforet
NAME                           CHART VERSION APP VERSION DESCRIPTION
(omit)
cloudforet/spaceone             1.11.11       1.11.11     A Helm chart for SpaceONE
(omit)
```

### Install cloudforet (refer to : https://github.com/cloudforet-io/charts)
```
$ helm install cloudforet cloudforet/spaceone -n spaceone
```

**Comment by GeunjeLEE** [comment]
This discussion has been resolved.
